tests/phpunit/CRM/Activity/BAO/ActivityTest.php

```diff
@@ -1135,4 +1135,140 @@ public function testSendEmailWithCampaign() {
+
+    $outBoundSmsActivityId = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'SMS');
+    $activityStatusCompleted = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', 'Completed');
+    $details = 'createSendSmsTest text';
+    $this->assertEquals($activity['activity_type_id'], $outBoundSmsActivityId, 'Wrong activity type is set.');
+    $this->assertEquals($activity['status_id'], $activityStatusCompleted, 'Expected activity status Completed.');
+    $this->assertEquals($activity['subject'], 'createSendSmsTest subject', 'Activity subject does not match.');
+    $this->assertEquals($activity['details'], $details, 'Activity details does not match.');
+    $this->assertEquals("Recipient phone number is invalid or recipient does not want to receive SMS", $sent[0]->message, "Expected error doesn't match");
+    $this->assertEquals(0, $success, "Expected success to be 0");
+  }
+
+  public function testSendSmsMobilePhoneNumber() {
+    list($sent, $activityId, $success) = $this->createSendSmsTest(2);
+    $activity = $this->civicrm_api('activity', 'getsingle', array('id' => $activityId, 'version' => $this->_apiversion));
+
+    $outBoundSmsActivityId = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'SMS');
+    $activityStatusCompleted = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', 'Completed');
+    $details = 'createSendSmsTest text';
+    $this->assertEquals($activity['activity_type_id'], $outBoundSmsActivityId, 'Wrong activity type is set.');
+    $this->assertEquals($activity['status_id'], $activityStatusCompleted, 'Expected activity status Completed.');
+    $this->assertEquals($activity['subject'], 'createSendSmsTest subject', 'Activity subject does not match.');
+    $this->assertEquals($activity['details'], $details, 'Activity details does not match.');
+    $this->assertEquals(TRUE, $sent, "Expected sent should be true");
+    $this->assertEquals(1, $success, "Expected success to be 1");
+  }
+
+  /**
+   * @param int $phoneType (0=no phone, phone_type option group (1=fixed, 2=mobile)
+   */
+  public function createSendSmsTest($phoneType = 0) {
+    require_once 'CiviTest/CiviTestSMSProvider.php';
+    // Create an SMS provider "CiviTestSMSProvider". Civi handles "CiviTestSMSProvider" as a special case and allows it to be instantiated
+    //  in CRM/Sms/Provider.php even though it is not an extension.
+    $groupID = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionGroup', 'sms_provider_name', 'id', 'name');
+    $params = array(
+      'option_group_id' => $groupID,
+      'label' => 'unittestSMS',
+      'value' => 'unit.test.sms',
+      'name'  => 'CiviTestSMSProvider',
+      'is_default' => 1,
+      'is_active'  => 1,
+      'version'    => 3,
+    );
+    $providerValue = civicrm_api3('option_value', 'create', $params);
```

Comment:
@mattwire This option value isn't properly being torn down me thinks following the test causing the test failures